src/model_config_utils.cc

```diff
@@ -784,6 +784,30 @@ GetNormalizedModelConfig(
   return Status::Success;
 }
 
+Status
+SetDefaultInstanceCount(
+    ::inference::ModelInstanceGroup* group, const std::string& backend)
+{
+  const int default_cpu_instance_count = 2;
+  if (group->kind() == inference::ModelInstanceGroup::KIND_CPU) {
+    // Backends opt into the default_cpu_instance_count since
+    // some backends (pytorch, OpenVINO) don't perform well/have high overhead
+    // when using multiple instances.
+    bool use_default_cpu_instance_count = (backend == kTensorRTBackend) ||
+                                          (backend == kTensorFlowBackend) ||
+                                          (backend == kOnnxRuntimeBackend);
+    if (use_default_cpu_instance_count) {
+      group->set_count(default_cpu_instance_count);
+    } else {
+      group->set_count(1);
+    }
+  } else {
+    group->set_count(1);
+  }
```

Comment:
optional: can this code be cleaned up at all by just setting group count to 1 at the beginning, and setting to 2 `if use_default_cpu_instance_count` ?

Reply:
You are right, it can be cleaned up. Updated.